layers/dns.go

```diff
@@ -464,16 +465,32 @@ func (rr *DNSResourceRecord) String() string {
 	return "..."
 }
 
+func decodeCharacterStrings(data []byte) ([][]byte, error) {
+	var strings [][]byte
```

Comment:
`decodeCharacterStrings` will be called for TXT and HINFO records.

HINFO records contain exactly 2 strings, but HINFO does not appear in practice; FWIW, [wikipedia](https://en.wikipedia.org/w/index.php?title=List_of_DNS_record_types&diff=prev&oldid=213247738) claims they are not in current use since 2008.

TXT records may contain more than one string, but almost never do. So I'd say that `make([][]byte, 0, 1)` would be more practical; but this, I believe, performs the same number of allocations as `make([][]byte)` followed by one `append(strings, slice)`, so maybe this line of code should remain as is?